### DISCUSSION TITLE
An Update and a Thank You

### DISCUSSION THREAD

**Discussion by navarr** [wardenenv/warden#599] [Announcements]
The first order of business is a slippage in our goal for Warden 1.0.  I initially set a goal of this for March 1st, but I am moving this to May 1st.  This date slippage is due to complications that I had in securing sponsorship for time to contribute, but I am happy to say that we have secured said sponsorship and work will pick up pace.

One of the primary issues in releasing 1.0 with full ARM64 support has been the Alpine system we have been hoping to import from Den. This is one of two competing approaches, with the other being a Debian-based system.  Since this decision has not been completely finalized, I would love to her your thoughts on Alpine vs. Debian in the thread.

And that brings us to one of the most important parts of this announcement: A **HUGE** round of thank yous.

The two most major shout-outs would be [SwiftOtter](https://www.swiftotter.com/) who has been willing to sponsor some of my time on this project and who created the Den fork which will be a huge contributor to adding AMD64 support; and [SanSec](https://www.sansec.io/) which is our largest fiscal sponsor.  Because of SanSec and our other backers (@talesh, @customgento, and Adam Paterson) we've hit both of our monthly fiscal goals within only a few months.  Assuming no decrease in contributions, we plan to:

1. Renew the `warden.dev` domain name for our landing page and documentation hosting ($15/m goal)
2. Incorporate and file for 501(c)6 status before the 2024 tax year (+$110/m goal)

While Goal 1 is important, Goal 2 is a huge additional undertaking. It's important to do to prevent taxation of our contributions.  Provided we achieve 501(c)6 status, Warden will no longer be taxed 21% of contributions after costs.  That's a **large** number that I am eager to reduce to 0%.  Unfortunately, we will not quality for a 501(c)3 status, so any contributions to Warden will not be deductible from personal income taxes stateside.

That's why I cannot express enough gratitude to our fiscal and time sponsors for making the continuation of this project feasible.

Thank you for reading, and I hope you look forward to 1.0!

**Comment by navarr** [comment]
Also a quick reminder that the finances for Warden are, much like the project itself, _open source_.

You can view all our transactions at https://opencollective.com/warden

**Reply by navarr** [reply]
Also, an earlier version of this post said "**501(c)8**". That was a mistake on my part, the actual tax classification we'd be shooting for is a **501(c)6**.

**Comment by bap14** [comment]
I'd be in support of having Debian-based images. Either migrating to them entirely, or just supporting both Alpine and Debian.

**Reply by navarr** [reply]
My only concern w/r/t migrating to them entirely would be killing BC-compatibility with CentOS.  Any scripts written for CentOS wouldn't be backwards compatible if we switch to Debian.

That said, We could do a legacy CentOS / Debian system similar to what I was thinking of doing with CentOS / Alpine, where to switch to the "new" images you have to include a flag in your .env

**Reply by bap14** [reply]
What scripts are you referring to? (I haven't dived (dove?, doven?, diven?) too deep into Warden so not sure if you're talking about core scripts or user-supplied scripts). Outside of some differences like apk/apt/yum (and security related items between CentOS/RockyLinux and Debian and Alpine) I can't think of a reason a sensible BaSH script wouldn't port to Debian.

**Reply by navarr** [reply]
User-supplied scripts.  Any move like that can cause issues just from the differences of pre-installed software etc and ideally I'd like to avoid that as the _default_, but make it opt-in with a .env flag and supply that flag for new env-init scripts
